airflow-core/tests/unit/api_fastapi/auth/managers/simple/test_middleware.py

```diff
@@ -0,0 +1,172 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import pytest
+from fastapi.testclient import TestClient
+
+from airflow.api_fastapi.app import create_app
+
+from tests_common.test_utils.config import conf_vars
+
+pytestmark = pytest.mark.db_test
+
+
+@pytest.fixture
+def all_access_test_client():
+    with conf_vars(
+        {
+            ("core", "simple_auth_manager_all_admins"): "true",
+            ("webserver", "expose_config"): "true",
+        }
+    ):
+        app = create_app()
+        yield TestClient(app)
+
+
+@pytest.mark.parametrize(
+    "method, path",
+    [
+        ("GET", "/api/v1/{_:path}"),
+        ("GET", "/api/v2/assets"),
+        ("GET", "/api/v2/assets/aliases"),
+        ("GET", "/api/v2/assets/aliases/{asset_alias_id}"),
+        ("GET", "/api/v2/assets/events"),
+        ("POST", "/api/v2/assets/events"),
+        ("GET", "/api/v2/assets/{asset_id}"),
+        ("POST", "/api/v2/assets/{asset_id}/materialize"),
+        ("GET", "/api/v2/assets/{asset_id}/queuedEvents"),
+        ("DELETE", "/api/v2/assets/{asset_id}/queuedEvents"),
+        ("GET", "/api/v2/backfills"),
+        ("POST", "/api/v2/backfills"),
```

Comment:
I don't think we need to enumerate all of the endpoints.

Reply:
Tests were failing. This was just a experiment to see which endpoint was causing it. I'll be iterating on the app.routes